backend/src/main/java/ai/verta/modeldb/configuration/AppConfigBeans.java

```diff
@@ -0,0 +1,412 @@
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import javax.annotation.PreDestroy;
+import javax.servlet.Filter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties({FileStorageProperties.class})
+public class AppConfigBeans {
+  private static final Logger LOGGER = LogManager.getLogger(AppConfigBeans.class);
+  private final AppContext appContext;
+  protected Optional<Server> server = Optional.empty();
+
+  // metric for prometheus monitoring
+  private static final Gauge up =
+      Gauge.build()
+          .name("verta_backend_up")
+          .help("Binary signal indicating that the service is up and working.")
+          .register();
+
+  public AppConfigBeans(AppContext appContext) {
+    this.appContext = appContext;
+  }
+
+  @Bean
+  public Filter webMvcTracingFilter(OpenTelemetry openTelemetry) {
+    return SpringWebMvcTelemetry.builder(openTelemetry).build().newServletFilter();
+  }
+
+  @Bean
+  public OpenTelemetry openTelemetry(Config config) {
+    return config.getOpenTelemetry();
+  }
+
+  @Bean
+  public S3Service getS3Service() throws ModelDBException, IOException {
+    String bucketName = System.getProperty(ModelDBConstants.CLOUD_BUCKET_NAME);
+    if (bucketName != null && !bucketName.isEmpty()) {
+      return new S3Service(System.getProperty(ModelDBConstants.CLOUD_BUCKET_NAME));
+    }
+    return null;
+  }
+
+  @Bean
+  public MDBConfig config() {
+    var config = MDBConfig.getInstance();
+    App.getInstance().mdbConfig = config;
+
+    // Configure spring HTTP server
+    LOGGER.info("Configuring spring HTTP traffic on port: {}", config.getSpringServer().getPort());
+    System.getProperties().put("server.port", config.getSpringServer().getPort());
+
+    return config;
+  }
+
+  @Bean
+  Executor executor(Config config) {
+    // Initialize executor so we don't lose context using Futures
+    return FutureGrpc.initializeExecutor(config.getGrpcServer().getThreadCount());
+  }
+
+  @Bean
+  public JdbiUtil jdbiUtil(MDBConfig config) throws SQLException, InterruptedException {
+    return JdbiUtil.getInstance(config);
+  }
+
+  @Bean
+  ServiceSet serviceSet(MDBConfig config, ArtifactStoreService artifactStoreService)
+      throws IOException {
+    // Initialize services that we depend on
+    return ServiceSet.fromConfig(config, artifactStoreService);
+  }
+
+  @Bean
+  public DAOSet daoSet(MDBConfig config, ServiceSet services, Executor handleExecutor) {
+    if (config.isMigration()) {
+      return null;
+    }
+
+    // Initialize data access
```

Comment:
some of these comments are redundant, as it's clear what's happening from the code.

Reply:
Resolved